k8s/agones/schema.yaml

```diff
@@ -0,0 +1,128 @@
+      serviceAccount:
+        roles:
+        - type: ClusterRole
+          rulesType: CUSTOM
+          rules:
+          - apiGroups: [""]
+            resources: ["events"]
+            verbs: ["create", "patch"]
+          - apiGroups: [""]
+            resources: ["pods"]
+            verbs: ["create", "delete", "list", "watch"]
+          - apiGroups: [""]
+            resources: ["nodes"]
+            verbs: ["list", "watch"]
+          - apiGroups: ["apiextensions.k8s.io"]
+            resources: ["customresourcedefinitions"]
+            verbs: ["get"]
+          - apiGroups: ["stable.agones.dev"]
+            resources: ["gameservers", "gameserversets"]
+            verbs: ["create", "delete", "get", "list", "update", "watch"]
+          - apiGroups: ["stable.agones.dev"]
+            resources: ["gameservers"]
+            verbs: ["patch"]
+          - apiGroups: ["stable.agones.dev"]
+            resources: ["fleets", "fleetallocations", "fleetautoscalers"]
+            verbs: ["get", "list", "update", "watch"]
+          - apiGroups: ["stable.agones.dev"]
+            resources: ["gameserverallocations"]
+            verbs: ["list", "watch",  "delete"]
+
+  agones.agones.serviceaccount.sdk:
+    type: string
+    title: Service account used by Agones sdk controller.
+    x-google-marketplace:
+      type: SERVICE_ACCOUNT
+      serviceAccount:
+        roles:
+        - type: ClusterRole
+          rulesType: CUSTOM
+          rules:
+          - apiGroups: [""]
+            resources: ["events"]
+            verbs: ["create"]
+          - apiGroups: ["stable.agones.dev"]
+            resources: ["gameservers"]
+            verbs: ["list", "update", "watch"]
+
+  # Images references don't show in the UI and are used to track required
+  # binaries for deployment
+  marketplace_integration.image:
+    type: string
+    x-google-marketplace:
+      type: DEPLOYER_IMAGE
+
+  AGONES_CONTROLLER_IMAGE:
+    type: string
+    default: $REGISTRY:$TAG
+    x-google-marketplace:
+      type: IMAGE
+      image:
+        generatedProperties:
+          splitToRegistryRepoTag:
+            registry: agones.agones.image.registry
+            repo: agones.agones.image.controller.name
+            tag: agones.agones.image.controller.tag
+
+  AGONES_SDK_IMAGE:
+    type: string
+    default: $REGISTRY/agones-sdk:$TAG
+    x-google-marketplace:
+      type: IMAGE
+      image:
+        generatedProperties:
+          splitToRegistryRepoTag:
+            registry: agones.agones.image.registry
+            repo: agones.agones.image.sdk.name
+            tag: agones.agones.image.sdk.tag
+
+  AGONES_PING_IMAGE:
+    type: string
+    default: $REGISTRY/agones-ping:$TAG
+    x-google-marketplace:
+      type: IMAGE
+      image:
+        generatedProperties:
+          splitToRegistryRepoTag:
+            registry: agones.agones.image.registry
+            repo: agones.agones.image.ping.name
+            tag: agones.agones.image.ping.tag
+
+required:
```

Comment:
Can you perhaps explain what required actually means? Is it that the user has to provide? I removed that from the images since those don't show up in the UI, and are provided systematically. Hence I didn't think "required" would make sense in that context.

Reply:
Sure, all properties will default values don't have to be inside the required list.